app/services/notification_service.rb

```diff
@@ -0,0 +1,31 @@
+class NotificationService
+  include ActiveModel::Validations
+
+  def initialize
+    OneSignal::OneSignal.api_key = ENV['PUSH_API_KEY']
+  end
+
+  def notify(push_token, message, data)
+    @push_token = push_token
+    @message = message
+    @data = data
+    OneSignal::Notification.create(params: params)
+  end
+
+  def all_notifications
+    OneSignal::Notification.all(params: { app_id: ENV['PUSH_APP_ID'] })
```

Comment:
What is this number? Because it's not the notifications for a specific device, right? Why do you need to use it?

Reply:
Right. Because I want to retrieve all notifications from my 'Target app', and OneSignal's API requires _app_id_ to be sent as a parameter.
Look at this: [https://documentation.onesignal.com/reference#view-notifications](https://documentation.onesignal.com/reference#view-notifications)

Reply:
But why do you want to retrieve all the notifications?

Reply:
Because at `spec/requests/api/v1/spots/create_spec.rb` at `sends the matched user a notification` I want to test that after creating a notification, it actually creates one at OneSignal app. I don't know if this is the best way to do it.

Reply:
You shouldn't add methods in the service if you only use it in the test. You should find another way to test it.

Reply:
You shouldn't add methods in the service if you only use it in the test.